firstvoices/backend/management/commands/copy_site.py

```diff
@@ -0,0 +1,581 @@
+        if force_delete:
+            Site.objects.filter(slug=target_slug).delete()
+        else:
+            raise AttributeError(
+                f"Site with slug {target_slug} already exists. Use --force-delete to override."
+            )
+
+
+def create_new_site(source_site, target_site_slug, user):
+    new_site = Site(
+        slug=target_site_slug,
+        title=target_site_slug,
+        language=source_site.language,
+        visibility=source_site.visibility,
+        is_hidden=source_site.is_hidden,
+        created_by=user,
+        last_modified_by=user,
+    )
+
+    new_site.contact_users.clear()
+    new_site.save()
+    return new_site
+
+
+def copy_site_features(source_site, target_site):
+    site_features = list(SiteFeature.objects.filter(site=source_site))
+    for site_feature in site_features:
+        site_feature.id = uuid.uuid4()
+        site_feature.site = target_site
+        site_feature.save()
+
+
+def copy_all_characters_and_return_map(
+    source_site, target_site, audio_map, image_map, video_map
+):
+    character_map = {}
+    characters = list(Character.objects.filter(site=source_site))
+
+    for character in characters:
+        source_media = {
+            "audio": list(character.related_audio.all().values_list("id", flat=True)),
+            "images": list(character.related_images.all().values_list("id", flat=True)),
+            "videos": list(character.related_videos.all().values_list("id", flat=True)),
+        }
+
+        source_char_id = character.id
+        target_char_id = uuid.uuid4()
+        character_map[source_char_id] = target_char_id
+
+        character.id = target_char_id
+        character.site = target_site
+        character.save()
+
+        copy_related_media(character, source_media, audio_map, image_map, video_map)
+        character.save()
+
+    variants = list(CharacterVariant.objects.filter(site=source_site))
+    for variant in variants:
+        source_base_character = variant.base_character
+
+        variant.id = uuid.uuid4()
+        variant.site = target_site
+        variant.base_character_id = character_map[source_base_character.id]
+        variant.save()
+
+    ignored_characters = list(IgnoredCharacter.objects.filter(site=source_site))
+    for ignored_character in ignored_characters:
+        ignored_character.id = uuid.uuid4()
+        ignored_character.site = target_site
+        ignored_character.save()
+
+    return character_map
+
+
+def copy_alphabet(source_site, target_site):
+    alphabets = list(Alphabet.objects.filter(site=source_site))
+    for alphabet in alphabets:
+        alphabet.id = uuid.uuid4()
+        alphabet.site = target_site
+        alphabet.save()
+
+
+def copy_categories_and_return_map(source_site, target_site, user):
+    category_map = {}
+
+    # Removing auto-generated categories
+    Category.objects.filter(site=target_site).delete()
+    # Copy over all the parent categories
+    categories = list(
```

Comment:
[minor] maybe rename this variable parent_categories to better follow what is happening in the code